mcp/src/main/java/io/modelcontextprotocol/server/McpServerFeatures.java

```diff
@@ -203,52 +204,99 @@ record Sync(McpSchema.Implementation serverInfo, McpSchema.ServerCapabilities se
 	/**
 	 * Specification of a tool with its asynchronous handler function. Tools are the
 	 * primary way for MCP servers to expose functionality to AI models. Each tool
-	 * represents a specific capability, such as:
-	 * <ul>
-	 * <li>Performing calculations
-	 * <li>Accessing external APIs
-	 * <li>Querying databases
-	 * <li>Manipulating files
-	 * <li>Executing system commands
-	 * </ul>
-	 *
-	 * <p>
-	 * Example tool specification: <pre>{@code
-	 * new McpServerFeatures.AsyncToolSpecification(
-	 *     new Tool(
-	 *         "calculator",
-	 *         "Performs mathematical calculations",
-	 *         new JsonSchemaObject()
-	 *             .required("expression")
-	 *             .property("expression", JsonSchemaType.STRING)
-	 *     ),
-	 *     (exchange, args) -> {
-	 *         String expr = (String) args.get("expression");
-	 *         return Mono.fromSupplier(() -> evaluate(expr))
-	 *             .map(result -> new CallToolResult("Result: " + result));
-	 *     }
-	 * )
-	 * }</pre>
+	 * represents a specific capability.
 	 *
 	 * @param tool The tool definition including name, description, and parameter schema
-	 * @param call The function that implements the tool's logic, receiving arguments and
-	 * returning results. The function's first argument is an
-	 * {@link McpAsyncServerExchange} upon which the server can interact with the
-	 * connected client. The second arguments is a map of tool arguments.
+	 * @param call Deprecated. Uset he {@link AsyncToolCallSpecification#callTool}
+	 * instead.
+	 * @param callTool The function that implements the tool's logic, receiving a
+	 * {@link McpAsyncServerExchange} and a
+	 * {@link io.modelcontextprotocol.spec.McpSchema.CallToolRequest} and returning
+	 * results. The function's first argument is an {@link McpAsyncServerExchange} upon
+	 * which the server can interact with the connected client. The second arguments is a
+	 * map of tool arguments.
 	 */
 	public record AsyncToolSpecification(McpSchema.Tool tool,
-			BiFunction<McpAsyncServerExchange, Map<String, Object>, Mono<McpSchema.CallToolResult>> call) {
+			@Deprecated BiFunction<McpAsyncServerExchange, Map<String, Object>, Mono<McpSchema.CallToolResult>> call,
+			BiFunction<McpAsyncServerExchange, McpSchema.CallToolRequest, Mono<McpSchema.CallToolResult>> callTool) {
```

Comment:
NIT:

`handler` or `toolHandler` can be used for instead of `callTool`

Reply:
We are already using the `handler` as the parameter's name in `McpServer` builder.

- https://github.com/modelcontextprotocol/java-sdk/blob/main/mcp/src/main/java/io/modelcontextprotocol/server/McpServer.java#L325-L326

Reply:
Isn't callHandler a more appropriate name?